tests/resources/test_snippets.py

```diff
@@ -58,288 +67,177 @@
 
     assert resp.status == 200
     assert await resp.json() == []
 
 
-async def test_get_snippets(testapp, snippets):
-    resp = await testapp.get(
-        '/snippets',
-        headers={
-            'Accept': 'application/json',
-        })
-    assert resp.status == 200
-
-    # snippets must be returned in descending order (newer first)
-    expected = sorted(
-        snippets,
-        key=operator.itemgetter('created_at'),
-        reverse=True)
-
-    for snippet_db, snippet_api in zip(expected, await resp.json()):
-        _compare_snippets(snippet_db, snippet_api)
-
-
-async def test_get_snippets_filter_by_title(testapp, snippets):
-    resp = await testapp.get(
-        '/snippets?title=snippet+%231',
-        headers={
-            'Accept': 'application/json',
-        })
-    assert resp.status == 200
-    expected = [snippets[0]]
-    for snippet_db, snippet_api in zip(expected, await resp.json()):
-        _compare_snippets(snippet_db, snippet_api)
-
-    nonexistent = await testapp.get(
-        '/snippets?title=nonexistent',
-        headers={
-            'Accept': 'application/json',
-        })
-    assert nonexistent.status == 200
-    assert list(nonexistent.json()) == []
-
-    regexes_are_escaped = await testapp.get(
-        '/snippets?title=%5Esnippet.%2A',  # title=^snippet.*
-        headers={
-            'Accept': 'application/json',
-        })
-    assert regexes_are_escaped.status == 200
-    assert list(regexes_are_escaped.json()) == []
-
-
-async def test_get_snippets_filter_by_title_bad_request(testapp, snippets):
-    resp = await testapp.get(
-        '/snippets?title=',
-        headers={
-            'Accept': 'application/json',
-        })
-    assert resp.status == 400
-    assert await resp.json() == {
-        'message': '`title` - empty values not allowed.'
-    }
-
-
-async def test_get_snippets_filter_by_tag(testapp, snippets):
-    resp = await testapp.get(
-        '/snippets?tag=tag_c',
-        headers={
-            'Accept': 'application/json',
-        })
-    assert resp.status == 200
-    expected = [snippets[1]]
-    for snippet_db, snippet_api in zip(expected, await resp.json()):
-        _compare_snippets(snippet_db, snippet_api)
-
-    nonexistent = await testapp.get(
-        '/snippets?tag=nonexistent_tag',
-        headers={
-            'Accept': 'application/json',
-        })
-    assert nonexistent.status == 200
-    assert list(nonexistent.json()) == []
-
-
-@pytest.mark.parametrize(
-    'value',
-    ['', 'test%20tag'],
-    ids=['empty', 'whitespace']
-)
-async def test_get_snippets_filter_by_tag_bad_request(value, testapp, snippets):
-    resp = await testapp.get(
-        '/snippets?tag=' + value,
-        headers={
-            'Accept': 'application/json',
-        })
-    assert resp.status == 400
-    assert await resp.json() == {
-        'message': "`tag` - value does not match regex '[\\w_-]+'."
-    }
-
+@pytest.mark.parametrize('params, rv, link', [
```

Comment:
(here and below) maybe we should use the `ids` param to give these cases explicit names?